File: SYP.Web/Modules/Order/OrderStatusHist/Editor/OrderStatusHistEditorColumns.cs
namespace SYP.Order.Columns;

[ColumnsScript("Order.OrderStatusHistEditor")]
[BasedOnRow(typeof(OrderStatusHistRow), CheckNames = true)]
public class OrderStatusHistEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public long Id { get; set; }
    public string OrderNumber { get; set; }
    public int OldStatus { get; set; }
    public int NewStatus { get; set; }
    public string ChangedByUserUsername { get; set; }
    [EditLink]
    public string ChangedByUserRole { get; set; }
    public string ChangeReason { get; set; }
    public DateTime ChangeDate { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Order/OrderStatusHistEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderStatusHistRow } from "./OrderStatusHistRow";

export interface OrderStatusHistEditorColumns {
    Id: Column<OrderStatusHistRow>;
    OrderNumber: Column<OrderStatusHistRow>;
    OldStatus: Column<OrderStatusHistRow>;
    NewStatus: Column<OrderStatusHistRow>;
    ChangedByUserUsername: Column<OrderStatusHistRow>;
    ChangedByUserRole: Column<OrderStatusHistRow>;
    ChangeReason: Column<OrderStatusHistRow>;
    ChangeDate: Column<OrderStatusHistRow>;
}

export class OrderStatusHistEditorColumns extends ColumnsBase<OrderStatusHistRow> {
    static readonly columnsKey = 'Order.OrderStatusHistEditor';
    static readonly Fields = fieldsProxy<OrderStatusHistEditorColumns>();
}